Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -237,7 +241,7 @@ class ScopePusher {
     void set_contains_direct_call_to_eval()
     {
         m_contains_direct_call_to_eval = true;
-        m_can_use_local_variables = false;
+        m_can_use_local_or_global_variables = false;
```

Comment:
This name really feels a bit weird, since it's perfectly fine (literally speaking) to use both local and global variable in all of these contexts. What we're really talking about is "variable access must use environment" or something like that, right?

Reply:
yes, this flag indicates whether there is eval somewhere in a scope chain so I renamed this to `m_screwed_by_eval_in_scope_chain`.